src/entity/Living.php

```diff
@@ -237,15 +237,14 @@ public function setSwimming(bool $value = true) : void{
 		if($value !== $this->isSwimming()) {
 			$this->swimming = $value;
 			if($this->swimming){
-				$this->originalWidth = $this->width;
-				$this->originalHeight = $this->height;
-				$this->originalEyeHeight = $this->eyeHeight;
-				$this->height = $this->eyeHeight = $this->width;
+				$width = $this->size->getWidth();
+				$this->originalWidth = $width;
+				$this->originalHeight = $this->size->getHeight();
+				$this->originalEyeHeight = $this->size->getEyeHeight();
```

Comment:
why not just make an originalSize field? ...

Reply:
great idea dktapps